### PR TITLE
Echo

### PR DESCRIPTION
**Pull request by Arthi-chaud**
Coding style compliant

Echo builtin  + backstick and inhibitor management (not in final state, has only been coded for tests purposes)

### CODE REVIEW

File: src/args.c [hunk 1]
```diff
@@ -28,27 +30,67 @@ int get_argc(char **argv)
...
+            cmd[i + 1] = '\n';
+        if (cmd[i + 1] == 'b')
+            cmd[i + 1] = '\b';
+        if (cmd[i + 1] == 't')
+            cmd[i + 1] = '\t';
+        if (cmd[i + 1] == 'v')
+            cmd[i + 1] = '\v';
+        if (cmd[i + 1] != 'v' && cmd[i + 1] != 'n'
+        && cmd[i + 1] != 'b' && cmd[i + 1] != 't')
+            cmd[i] = cmd[i + 1];
```
Comment (Octopus773) [inline src/args.c]: you can do a switch with a default case to avoid if forests

File: src/builtin/builtin_echo.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    value = my_getenv(env->vars, isolated_var);
+    if (value) {
+        printf("%s", value);
+        free(isolated_var);
+        return;
+    }
+    printf("%s: Undefined variable.\n", isolated_var);
+    my_setenv(env->vars, "?", "1");
+    free(isolated_var);
```
Comment (Octopus773) [inline src/builtin/builtin_echo.c]: I think a anterior get_var function should be implemented

**Review by Octopus773 — APPROVED** [review]

**Review by GitBluub — APPROVED** [review]

**Review by Octopus773 — APPROVED** [review]

File: src/utility/replace_var.c [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    if (!isolated_var)
+        return (NULL);
+    if (isolated_var == cmd)
+        return (cmd);
+    for (; isolated_var[var_len] && isolated_var[var_len] != ' '
+    && isolated_var[var_len] != '\t'; var_len++);
+    if (!var_len && cmd[0] != '\"')
+        return (cmd);
+    value = get_var_value(isolated_var, env);
+    if (!value)
```
Comment (zoriya) [inline src/utility/replace_var.c]: It won't work if the var is not the only thing in the string "not$VAR" for example will return "" and not "not" on undefined VAR.

File: src/utility/replace_var.c [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+char * replace_var(char *cmd, env_t *env)
+{
+    char *isolated_var = NULL;
+    char *value = NULL;
+    char *new_cmd = NULL;
+    int var_len = 0;
+
+    if (cmd[0] == '\'')
+        return (cmd);
+    isolated_var = isolate_var(cmd, env);
```
Comment (zoriya) [inline src/utility/replace_var.c]: Your var handling will only work if there is one or less variables on a single string. Ex: "$VAR$VAR2" will parse $VAR but not $VAR2

File: src/utility/replace_var.c [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    for (int i = 0; cmd[i] && !isolated_var; i++)
+        if (cmd[i] == '$')
+            isolated_var = &cmd[i + 1];
+    if (!isolated_var)
+        return (cmd);
+    if (isolated_var[0] == 0)
+        return (cmd);
+    if ((isolated_var[0] == ' ' || isolated_var[0] == '\t' ||
+        !isolated_var[0]) && cmd[0] != '\"')
```
Comment (zoriya) [inline src/utility/replace_var.c]: Variable parsing should work without a " at the begining of the comand.

**Pull request closed without merging**